internal/provider/common/errors.go

```diff
@@ -54,9 +54,9 @@ func HandleCredentialError(ctx context.Context, invalidator environs.CredentialI
 		if invalidateErr != nil {
 			logger.Warningf(ctx, "could not invalidate stored cloud credential on the controller: %v", invalidateErr)
 		}
-		return true
+		return true, err
```

Comment:
What's the point of always returning the original input error?

Reply:
So you can chain it, which is often used in providers, not sure why.